src/pevm.rs

```diff
@@ -196,16 +201,23 @@ pub fn execute_revm<S: Storage + Send + Sync>(
     // to avoid "implicit" dependency among consecutive transactions that read & write there.
     // TODO: Refactor, improve speed & error handling.
     let beneficiary_values = mv_memory.consume_beneficiary();
+    let mut cumulative_gas_used: u128 = 0;
     Ok(execution_results
         .into_iter()
         .zip(beneficiary_values)
         .map(|(mutex, value)| {
-            let mut result_and_state = mutex.into_inner().unwrap().unwrap();
-            result_and_state.state.insert(
+            let ResultAndState { result, mut state } = mutex.into_inner().unwrap().unwrap();
+            state.insert(
                 beneficiary_address,
                 post_process_beneficiary(&mut beneficiary_account_info, value),
             );
-            result_and_state
+            cumulative_gas_used += result.gas_used() as u128;
```

Comment:
Deduplicate code with `execute_sequential`. Even better,  if we have a function taking in an iterator of `ResultAndState` and returns an iterator or vector of `PevmTxExecutionResult`, then we can separate output conversion with beneficiary post-processing & sequential execution.